src/libopensc/card-openpgp.c

```diff
@@ -709,40 +709,40 @@ pgp_parse_algo_attr_blob(sc_card_t *card, const pgp_blob_t *blob,
 
 int
 _pgp_handle_curve25519(sc_card_t *card,
-		sc_cardctl_openpgp_key_gen_store_info_t key_info, unsigned int do_num)
+		sc_cardctl_openpgp_key_gen_store_info_t *key_info, unsigned int do_num)
 {
-	if (!sc_compare_oid(&key_info.u.ec.oid, &curve25519_oid) &&
```

Comment:
given that we now pass the struct by reference, we should really make sure it is not NULL before dereferencing here.  The same in the other function.